tools/buildHeaders/header.cpp

```diff
@@ -123,6 +124,8 @@ namespace {
                                enumStyle_t style) const {
             return enumFmt(s, v, style, true);
         }
+        virtual std::string enumProvisionalStart(const std::string& name) const { return ""; };
+        virtual std::string enumProvisionalEnd(const std::string& name)  const { return ""; };
```

Comment:
```suggestion
        virtual std::string guardIfProvisionalStart(const std::string& name) const { return ""; };
        virtual std::string guardIfProvisionalEnd(const std::string& name)  const { return ""; };
```

just to convey better the function checks before emitting, otherwise it would be better to hoist the check out of the function